mediagrains/utils/grain_wrapper.py

```diff
@@ -0,0 +1,67 @@
+#
+# Copyright 2019 British Broadcasting Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""\
+Support for reading raw essence data and wrapping it in Grains, or reading Grains and returning essence data.
+This can be used for tasks like piping the output of ffmpeg (or similar) into a GSF file.
+"""
+import typing
+import copy
+
+from mediatimestamp.immutable import TimeRange
+
+from ..grain import GRAIN
+
+
+class GrainWrapper(object):
+    """Raw input and wrap it in Grains"""
+    def __init__(
+        self,
+        grain_constructor: typing.Callable,
```

Comment:
Should this be better annotated? `typing.Callable` is pretty vague

Reply:
I've reworked this based on our discussion to accept a Grain instead. How's 4c4dde7?